docs/howto/charm-to-poetry.rst

```diff
@@ -0,0 +1,107 @@
+.. _howto-migrate-to-poetry:
+
+Migrate from the ``charm`` plugin to Poetry
+===========================================
+
+Many charms use `Poetry`_ to manage their Python projects. For these charms, Charmcraft
+has a :ref:`craft_parts_poetry_plugin`. Migrating from the charm plugin provides some
+benefits, such as no longer having to maintain a ``requirements.txt`` file. If the
+charm to be migrated does not currently use poetry, refer to the
+`Poetry documentation <https://python-poetry.org/docs/basic-usage/>`_ for instructions
+on how to use poetry for a Python project.
+
+Update ``charmcraft.yaml``
+--------------------------
+
+The first step is to update ``charmcraft.yaml`` to include the correct parts definition.
+Depending on the history of a specific charm, it may not have an explicitly-included
+``parts`` section determining how to build the charm. In this case, a ``parts`` section
+can be created as follows:
+
+.. code-block:: yaml
+
+    parts:
+      my-charm:  # This can be named anything you want
```

Comment:
A morsel for thought: we generally [discourage the use of comments](https://docs.ubuntu.com/styleguide/en/#code-examples-in-documentation) in bash, but I'm not sure where configs fit in, since we encourage users to copy them. As a general principle it's better to use the doc to explain things.

Reply:
We use comments pretty heavily in our [config file templates](https://github.com/canonical/charmcraft/blob/main/charmcraft/templates/init-simple/charmcraft.yaml.j2). In this case, the goal of the comment is to counter something that's a fairly common misconception amongst charm authors (that the part name needs to be named the plugin name or the actual charm name) without having an aside about it, since that's out of scope for the document.